src/tracer.js

```diff
@@ -1,30 +1,24 @@
 'use strict'
 
+const semver = require('semver')
 const Tracer = require('./opentracing/tracer')
 const tags = require('../ext/tags')
+const scopes = require('../ext/scopes')
 
 const SPAN_TYPE = tags.SPAN_TYPE
 const RESOURCE_NAME = tags.RESOURCE_NAME
 const SERVICE_NAME = tags.SERVICE_NAME
 const ANALYTICS = tags.ANALYTICS
+const ASYNC_HOOKS = scopes.ASYNC_HOOKS
+const ASYNC_LISTENER = scopes.ASYNC_LISTENER
+const NOOP = scopes.NOOP
 
 class DatadogTracer extends Tracer {
   constructor (config) {
     super(config)
 
-    let ScopeManager
-    let Scope
-
-    if (process.env.DD_CONTEXT_PROPAGATION === 'false') {
-      ScopeManager = require('./scope/noop/scope_manager')
-      Scope = require('./scope/new/base')
-    } else {
-      ScopeManager = require('./scope/scope_manager')
-      Scope = require('./scope/new/scope')
-    }
-
-    this._scopeManager = new ScopeManager()
-    this._scope = new Scope()
+    this._scopeManager = getScopeManager(config)
```

Comment:
`getScopeManager()` doesn't (currently) take a `config`

Reply:
This was done to not have to worry about if it uses the config or not. I can remove it if you prefer since it's not used.

Reply:
The config is now used, so this issue no longer applies.